Source/AGXUnreal/Public/Terrain/AGX_MovableTerrainComponent.h

```diff
@@ -0,0 +1,300 @@
+#pragma once
+
+// AGX Dynamics for Unreal include.s
+#include "AGX_NativeOwner.h"
+#include "Terrain/TerrainBarrier.h"
+#include "Terrain/AGX_TerrainMeshUtilities.h"
+#include "AGX_ShovelReference.h"
+
+#include "CoreMinimal.h"
+#include "ProceduralMeshComponent.h"
+#include "AGX_MovableTerrainComponent.generated.h"
+
+
+class UAGX_ShapeComponent;
+class UAGX_TerrainMaterial;
+class UAGX_ShovelComponent;
+class UNiagaraSystem;
+class UNiagaraComponent;
+
+struct MeshTile
+{
+	FVector2D Center;
+	FVector2D Size;
+	FIntVector2 Resolution;
+	MeshTile(FVector2D TileCenter, FVector2D TileSize, FIntVector2 TileRes)
+	{
+		Center = TileCenter;
+		Size = TileSize;
+		Resolution = TileRes;
+	}
+};
+
+/**
+ *
+ */
+UCLASS(
+	ClassGroup = "AGX_Terrain", meta = (BlueprintSpawnableComponent),
+	Hidecategories = (Cooking, Collision, Input, LOD, Physics, Replication))
+class AGXUNREAL_API UAGX_MovableTerrainComponent : public UProceduralMeshComponent
```

Comment:
I think this should also derrive from `IAGX_NativeOwner` to make the Native barrier object survive Blueprint Reconstruction, i.e. make sure the new Movable Terrain that is created at blueprint reconstruction takes over ownership of the Native object which is not handled automatically.

What happens is that if you have a MovableTerrainComponent in a Blueprint, then instantiate that in the Level, and press Play, now if you change any properties in the Details panel, a Blueprint reconstruction will occur, which will destroy the original MovableTerrain and create a new one, and BeginPlay is then called on the new one.

This corresponds to the "Blueprint reconstruction support" checkbox in the MR description.

Ping me or Martin if you need more information/help, this is not trivial stuff.